temporian/core/operators/cast.py

```diff
@@ -0,0 +1,128 @@
+        if to not in ALL_TYPES:  # Will be 'not isinstance(to, DType)'
+            for origin_key in to:
+                if (
+                    origin_key not in ALL_TYPES  # Also here
+                    and origin_key not in event.feature_names
+                ):
+                    raise ValueError(f"Invalid key to cast: {origin_key}")
+
+        # Convert any input format to feature_name->target_dtype
+        target_dtypes = {}
+        for feature in event.features:
+            if to in ALL_TYPES:  # isinstance(to, DType):
+                target_dtypes[feature.name] = to
+            elif feature.name in to:
+                target_dtypes[feature.name] = to[feature.name]
+            elif feature.dtype in to:
+                target_dtypes[feature.name] = to[feature.dtype]
+            else:
+                target_dtypes[feature.name] = feature.dtype
+        self.add_attribute("target_dtypes", target_dtypes)
+        self.add_attribute("check_overflow", check_overflow)
+
+        # inputs
+        self.add_input("event", event)
+
+        # outputs
+        output_features = []
+        reuse_event = True
+        for feature in event.features:
+            if target_dtypes[feature.name] is feature.dtype:
+                # Reuse feature
+                output_features.append(feature)
+            else:
+                # Create new feature
+                reuse_event = False
+                output_features.append(
+                    # Note: we're not renaming output features here
+                    Feature(
+                        feature.name,
+                        target_dtypes[feature.name],
+                        feature.sampling,
+                        creator=self,
+                    )
+                )
+
+        # Output event: don't create new if all features are reused
+        self.add_output(
+            "event",
+            event
+            if reuse_event
+            else Event(
+                features=output_features,
+                sampling=event.sampling,
+                creator=self,
+            ),
+        )
+
+        self.check()
+
+    @classmethod
+    def build_op_definition(cls) -> pb.OperatorDef:
+        return pb.OperatorDef(
+            key="CAST",
+            attributes=[
+                pb.OperatorDef.Attribute(
+                    key="target_dtypes",
+                    type=pb.OperatorDef.Attribute.Type.MAP_STR_STR,
+                    is_optional=False,
+                ),
+                pb.OperatorDef.Attribute(
+                    key="check_overflow",
+                    type=pb.OperatorDef.Attribute.Type.BOOL,
+                    is_optional=False,
+                ),
+            ],
+            inputs=[
+                pb.OperatorDef.Input(key="event"),
+            ],
+            outputs=[pb.OperatorDef.Output(key="event")],
+        )
+
+
+operator_lib.register_operator(CastOperator)
+
+
+def cast(
```

Comment:
Could we have a docstring here explaining the function? This is a Public API function.